CMakeLists.txt

```diff
@@ -72,50 +74,82 @@ set(libebml_PUBLIC_HEADERS
 
 set(libebml_C_PUBLIC_HEADERS ebml/c/libebml_t.h)
 
-add_library(ebml ${libebml_SOURCES} ${libebml_PUBLIC_HEADERS} ${libebml_C_PUBLIC_HEADERS})
-if(WIN32)
-  include(CheckIncludeFile)
-  check_include_file(winapifamily.h HAVE_WINAPIFAMILY_H)
-  if(HAVE_WINAPIFAMILY_H)
-    target_compile_definitions(ebml PUBLIC HAVE_WINAPIFAMILY_H)
-  endif()
-endif()
-set_target_properties(ebml PROPERTIES
-  VERSION 5.0.0
-  SOVERSION 5
-  CXX_VISIBILITY_PRESET hidden
-  VISIBILITY_INLINES_HIDDEN ON
-)
-target_include_directories(ebml
-  PRIVATE
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
-  PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
-if(MSVC)
-  target_compile_definitions(ebml PRIVATE _CRT_SECURE_NO_WARNINGS)
-endif()
 
 include(GenerateExportHeader)
-generate_export_header(ebml EXPORT_MACRO_NAME EBML_DLL_API)
-target_sources(ebml
-  PRIVATE
-    ${CMAKE_CURRENT_BINARY_DIR}/ebml_export.h
-)
-
-if(NOT BUILD_SHARED_LIBS)
-  target_compile_definitions(ebml PUBLIC EBML_STATIC_DEFINE)
+foreach (TYPE IN ITEMS STATIC SHARED)
+  if (NOT DISABLE_${TYPE}_LIBS)
+
+    set(type_suffix "")
+    if ("${TYPE}" STREQUAL "STATIC")
+      string(TOLOWER "${TYPE}" type)
+      set(type_suffix "-${type}")
+    endif()
+
+    add_library(ebml${type_suffix} ${TYPE}
+      ${libebml_SOURCES}
+      ${libebml_PUBLIC_HEADERS}
+      ${libebml_C_PUBLIC_HEADERS})
+
+    if(WIN32)
+      include(CheckIncludeFile)
+      check_include_file(winapifamily.h HAVE_WINAPIFAMILY_H)
+      if(HAVE_WINAPIFAMILY_H)
+        target_compile_definitions(ebml${type_suffix} PUBLIC HAVE_WINAPIFAMILY_H)
+      endif()
+    endif()
+
+    target_include_directories(ebml${type_suffix}
+      PRIVATE
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+      PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+
+    generate_export_header(ebml${type_suffix} EXPORT_MACRO_NAME EBML_DLL_API BASE_NAME ebml)
+    target_sources(ebml${type_suffix}
+      PRIVATE
+        ${CMAKE_CURRENT_BINARY_DIR}/ebml_export.h)
+
+    if(MSVC)
+      target_compile_definitions(ebml${type_suffix} PRIVATE _CRT_SECURE_NO_WARNINGS)
+    endif()
+  endif()
+endforeach()
+
+if(NOT DISABLE_SHARED_LIBS)
+  set_target_properties(ebml PROPERTIES
+    VERSION 5.0.0
+    SOVERSION 5
+    OUTPUT_NAME ebml
+    DEFINE_SYMBOL "EBML_DLL_EXPORT"
+    VISIBILITY_INLINES_HIDDEN ON
+    C_VISIBILITY_PRESET hidden
+    CXX_VISIBILITY_PRESET hidden)
+
+  target_compile_definitions(ebml
+      PUBLIC EBML_DLL
+      PRIVATE EBML_DLL_EXPORT)
```

Comment:
Needs to be  `ebml_EXPORTS`, not `EBML_DLL_EXPORT`.

Reply:
The [libmatroska version](https://github.com/Matroska-Org/libmatroska/pull/53) doesn't have this and it's exporting the DLL symbols correctly. Removing this `target_compile_definitions` completely also fixes the issue.